dissect/target/filesystems/zip.py

```diff
@@ -0,0 +1,136 @@
+
+log = logging.getLogger(__name__)
+
+
+class ZipFilesystem(Filesystem):
+    """This Zip filesystem implementation is WIP and
+    has only been tested for opening java jar archives.
+    Its structure has been copied from the tar filesystem
+    implementation.
+    """
+
+    __fstype__ = "zip"
+
+    def __init__(self, fh, base=None, case_sensitive=True, zipinfo=None, *args, **kwargs):
+        fh.seek(0)
+
+        self.zip = zipfile.ZipFile(fh, mode="r")
+        self.base = base.strip("/") if base else ""
+
+        self._fs = VirtualFilesystem(case_sensitive=case_sensitive)
+
+        for member in self.zip.infolist():
+
+            if member.is_dir():
+                continue
+
+            mname = member.filename.strip("/")
+            if not mname.startswith(self.base):
+                continue
+
+            rel_name = mname[len(self.base) :]
+
+            # NOTE: Normally we would check here if the member is a symlink or not,
+            # however Python does not have symlink support in the zipfile module.
+            # see https://github.com/python/cpython/issues/82102
+
+            file_entry = ZipFilesystemEntry(self, rel_name, member)
+            self._fs.map_file_entry(rel_name, file_entry)
+
+        super().__init__(case_sensitive=case_sensitive, *args, **kwargs)
+
+    @staticmethod
+    def detect(fh):
+        """Detect a zip file on a given file-like object."""
+        offset = fh.tell()
+        try:
+            fh.seek(0)
+            zipfile.ZipFile(fileobj=fh, mode="r")
+            return True
+        except Exception as e:
+            log.warning("Failed to detect zip filesystem", exc_info=e)
+            return False
+        finally:
+            fh.seek(offset)
+
+    def get(self, path):
+        """Returns a ZipFilesystemEntry object corresponding to the given path."""
+        return self._fs.get(path)
+
+
+class ZipFilesystemEntry(VirtualFile):
+    def _resolve(self):
+        # NOTE: Normally we would check here if the member is a symlink or not,
+        # however Python does not have symlink support in the zipfile module.
+        # see https://github.com/python/cpython/issues/82102
+        return self
+
+    def open(self):
+        """Returns file handle (file-like object)."""
+        try:
+            f = self.fs.zip.read(self.entry)
+            return BufferedStream(io.BytesIO(f), size=self.entry.file_size)
+        except Exception as e:
+            print(str(e))
+            raise FileNotFoundError()
+
+    def is_dir(self):
+        """Return whether this entry is a directory. Resolves symlinks when possible."""
+        try:
+            return self._resolve().entry.isdir()
+        except FilesystemError:
+            return False
+
+    def is_file(self):
+        """Return whether this entry is a file. Resolves symlinks when possible."""
+        try:
+            return self._resolve().entry.isfile()
+        except FilesystemError:
+            return False
```

Comment:
```suggestion
        try:
            return not self.is_dir()
        except FilesystemError:
            return False
```
`zipfile.ZipInfo` has no `isfile`, so just use the inverse of `is_dir`. Note that an error should still result in `False`.